### PR TITLE
Convert test suite to specs

### LINKED ISSUE #21: Convert minitest suite to rspec
rspec is much more commonly used across the WeTransfer landscape, so we should use the tool that most people are comfortable with.

### PR DESCRIPTION
**Pull request by acroos**
## Description

It was flagged that many more people that may contribute to this project use rspec, so we should also use it.  It also turned out to be much easier for testing the connection adapter.  So a double-win!

Fixes #21 

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

- [x] My code follows the style guidelines set by rubocop
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: ghost_adapter.gemspec [hunk 1]
```diff
@@ -23,11 +23,12 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
 
   spec.add_dependency 'activerecord', '>= 5'
+  spec.add_dependency 'mysql2', '>= 0.4.0', '< 0.6.0'
```
Comment (nickdowse) [inline ghost_adapter.gemspec:26]: 👍

File: .rubocop.yml [hunk 2]
```diff
@@ -10,6 +10,7 @@ Style/RescueModifier:
 Metrics/BlockLength:
   Exclude:
     - test/**/*_test.rb
```
Comment (nickdowse) [inline .rubocop.yml:12]: Do we still need to exclude this?

**Review by nickdowse — APPROVED** [review]

File: spec/ghost_adapter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+        config = described_class.config.compact
+        expect('second').to eq config[keys[0]]
+        expect('first').to eq config[keys[1]]
+      end
+    end
+  end
+end
+
+#   def test_setup_called_twice
```
[the remark this answers is not in the document]
Comment (acroos) [reply spec/ghost_adapter_spec.rb:94]: oops

File: .rubocop.yml (hunk 2 above)
Comment (acroos) [reply .rubocop.yml:12]: nope!  good catch :)

PS there might be more minitest remnants, I think I got them all but not sure

**Pull request merged**